core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -19,6 +19,7 @@
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
```

Comment:
Please check and handle vectorCollector also if required.

Reply:
VectorCollector is not required for now. It will be handled in next Complex data type enhancement PR